flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adapter/ExecutionGraphToSchedulingTopologyAdapterTest.java

```diff
@@ -0,0 +1,188 @@
+
+	private ExecutionGraph executionGraph;
+
+	private ExecutionGraphToSchedulingTopologyAdapter adapter;
+
+	@Before
+	public void setUp() throws Exception {
+		JobVertex[] jobVertices = new JobVertex[2];
+		int parallelism = 3;
+		jobVertices[0] = createNoOpVertex(parallelism);
+		jobVertices[1] = createNoOpVertex(parallelism);
+		jobVertices[1].connectNewDataSetAsInput(jobVertices[0], ALL_TO_ALL, BLOCKING);
+		jobVertices[0].setInputDependencyConstraint(ALL);
+		jobVertices[1].setInputDependencyConstraint(ANY);
+		executionGraph = createSimpleTestGraph(
+			new JobID(),
+			taskManagerGateway,
+			triggeredRestartStrategy,
+			jobVertices);
+		adapter = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
+	}
+
+	@Test
+	public void testConstructor() {
+		// implicitly tests order constraint of getVertices()
+		assertGraphEquals(executionGraph, adapter);
+	}
+
+	@Test
+	public void testGetResultPartition() {
+		for (ExecutionVertex vertex : executionGraph.getAllExecutionVertices()) {
+			for (Map.Entry<IntermediateResultPartitionID, IntermediateResultPartition> entry : vertex.getProducedPartitions().entrySet()) {
+				IntermediateResultPartition partition = entry.getValue();
+				SchedulingResultPartition schedulingResultPartition = adapter.getResultPartition(entry.getKey())
+					.orElseThrow(() -> new IllegalArgumentException("can not find partition " + entry.getKey()));
+
+				assertPartitionEquals(partition, schedulingResultPartition);
+			}
+		}
+	}
+
+	private static void assertGraphEquals(
+		ExecutionGraph originalGraph,
+		SchedulingTopology adaptedTopology) {
+
+		Iterator<ExecutionVertex> originalVertices = originalGraph.getAllExecutionVertices().iterator();
+		Iterator<SchedulingExecutionVertex> adaptedVertices = adaptedTopology.getVertices().iterator();
+
+		while (originalVertices.hasNext()) {
+			ExecutionVertex originalVertex = originalVertices.next();
+			SchedulingExecutionVertex adaptedVertex = adaptedVertices.next();
+
+			assertVertexEquals(originalVertex, adaptedVertex);
+
+			List<IntermediateResultPartition> originalConsumedPartitions = IntStream.range(0, originalVertex.getNumberOfInputs())
+				.mapToObj(originalVertex::getInputEdges)
+				.flatMap(Arrays::stream)
+				.map(ExecutionEdge::getSource)
+				.collect(Collectors.toList());
+			Collection<SchedulingResultPartition> adaptedConsumedPartitions = adaptedVertex.getConsumedResultPartitions();
+
+			assertPartitionsEquals(originalConsumedPartitions, adaptedConsumedPartitions);
+
+			Collection<IntermediateResultPartition> originalProducedPartitions = originalVertex.getProducedPartitions().values();
+			Collection<SchedulingResultPartition> adaptedProducedPartitions = adaptedVertex.getProducedResultPartitions();
+
+			assertPartitionsEquals(originalProducedPartitions, adaptedProducedPartitions);
+		}
+		if (adaptedVertices.hasNext()) {
+			fail("Number of adapted vertices exceeds number of original vertices.");
```

Comment:
if block can be replaced with

```suggestion
assertFalse("Number of adapted vertices exceeds number of original vertices.", adaptedVertices.hasNext());
```

Reply:
OK